cs130_front/src/components/views/ClassesPage/ClassesPage.js

```diff
@@ -11,42 +12,155 @@ import SearchBar from '../../Search/SearchBar';
 import Text from '../../Text/Text';
 import ListingCreator from '../../ListingCreator/ListingCreator'
 import UserList from '../../UserList/UserList';
+import { ROUTE, USER_SERVER_AUTH, COURSE_SERVER, INVITATION_SERVER } from '../../../Config';
 
 function ClassesPage(props) {
-  const items = [{name: "Wow Squad", groupId: 123}, {name: "CA Squad", groupId: 124}, {name: "DM Squad", groupId: 125}]
-  const user = {firstName: "Ethan", lastName: "Wow", id: 123}
-  const classes = [{name: "Discrete Mathematics", courseId: 1, listings: [{poster: "Edgar Garcia", school: "University of Nevada, Reno", description: "Looking for a group of Nevada kids to talk about bipartite graphs! Hit me up!"}, {poster: "Colin Kaepernick", school: "University of Nevada, Reno", description: "Looking for 2 more Nevada kids!", groupId: 5, groupName: "DM Squad"}]}, {name: "Computer Architecture", courseId: 2, listings: [{poster: "Glenn Reinman", school: "University of California, Los Angeles", description: "Looking for a group of students to workout with!"}]}]
-
-  const members = [
-    { name: "Fred Warner", school: "BYU", id: 54, discord: "AllProFred", email: "fred@gmail.com" },
-    { name: "Kyle Juszczyk", school: "Harvard", id: 44, discord: "JuiceCheck", email: "juice@gmail.com" }
-  ]
-  const currentGroup = { id: 123, name: "DM Squad", courseName: "Calculus", members: members, day: "friday", time: "4:30pm" }
-  
-  const [mainTitleState, setMainTitle] = useState(classes[0].name);
-  const [mainListingsDefault, setMainListingsDefault] = useState(classes[0].listings);
-  const [mainListings, setMainListings] = useState(classes[0].listings);
+  const [classes, setClasses] = useState([]);
+  const [mainTitleState, setMainTitle] = useState("");
+  const [mainClassId, setMainClassId] = useState(null);
+  const [mainListingsDefault, setMainListingsDefault] = useState([]);
+  const [mainListings, setMainListings] = useState([]);
+  const [mainStudyBuddies, setMainStudyBuddies] = useState([]);
+  const [allGroups, setAllGroups] = useState({});
+  const [mainGroups, setMainGroups] = useState([]);
   const [currentTab, setCurrentTab] = useState("Study Buddies");
   const [input, setInput] = useState('');
   const [invitedUser, setInvitedUser] = useState(null);
-  //inviter is used by including it as an optional element for each user row in the user list 
-  //show if invite to group button is clicked etc, then pass the user info from the row to the inviter
-  //and groups information for current user can be gotten from the groups endpoint
-  function handleGroupInvite(user, group){
-    //send request with user and group info
 
+  const userId = localStorage.getItem('userId');
+  const config = {
+    headers: {
+      'Authorization': `Bearer ${localStorage.getItem('accessToken')}`
+    }
+  }
+  
```

Comment:
This useEffect handles switching to the class that had a listing added. The new listing should populate. This is a better UI since previously we need to reload the page ourselves and choose the class again. 

```
  useEffect(() => {
    async function initCourses() {
      try {
        classes.forEach(course => {
          if (addedListing.course_id == course.courseId) {
            classClicked(course);
          }
        })
      } catch (err) {
        // Handle err here. Either ignore the error, or surface the error up to the user somehow.
      }
    }
    initCourses();
  }, [addedListing, classes]);
```